src/Components/Web/src/Forms/InputBase.cs

```diff
@@ -187,6 +187,7 @@ protected string CssClass
 
 
         /// <inheritdoc />
+        [MemberNotNull(nameof(EditContext), nameof(CascadedEditContext))]
```

Comment:
It's fine to have this since you added it, but in the ordinary case, it's not very useful. The compiler cannot deduce the component lifecycle so so it's hard for it know these properties are not-null when executing any other method on a component.